src/delete.ts

```diff
@@ -13,9 +13,20 @@ export function getVersionIds(input: Input): Observable<string[]> {
       input.owner,
       input.repo,
       input.packageName,
-      input.numOldVersionsToDelete,
+      input.numOldVersionsToDelete + input.minVersionsToKeep,
       input.token
-    ).pipe(map(versionInfo => versionInfo.map(info => info.id)))
+    ).pipe(
+      map(versionInfo => {
+        const numberVersionsToDelete =
+          versionInfo.length - input.minVersionsToKeep
+        return numberVersionsToDelete <= 0
+          ? []
+          : versionInfo
+              .filter(info => !input.ignoreVersions.test(info.version))
```

Comment:
What does this test method do?

Reply:
The test method checks for package versions against the regex provided for package versions to be ignored. If they match we filter them out.